pandas/tests/indexes/test_base.py

```diff
@@ -41,80 +41,65 @@
 from pandas.core.indexes.api import Index, MultiIndex
 from pandas.core.sorting import safe_sort
 from pandas.tests.indexes.common import Base
+from pandas.tests.indexes.conftest import indices_dict
 import pandas.util.testing as tm
 from pandas.util.testing import assert_almost_equal
 
 
 class TestIndex(Base):
     _holder = Index
 
-    def setup_method(self, method):
-        self.indices = dict(
-            unicodeIndex=tm.makeUnicodeIndex(100),
-            strIndex=tm.makeStringIndex(100),
-            dateIndex=tm.makeDateIndex(100),
-            periodIndex=tm.makePeriodIndex(100),
-            tdIndex=tm.makeTimedeltaIndex(100),
-            intIndex=tm.makeIntIndex(100),
-            uintIndex=tm.makeUIntIndex(100),
-            rangeIndex=tm.makeRangeIndex(100),
-            floatIndex=tm.makeFloatIndex(100),
-            boolIndex=Index([True, False]),
-            catIndex=tm.makeCategoricalIndex(100),
-            empty=Index([]),
-            tuples=MultiIndex.from_tuples(zip(["foo", "bar", "baz"], [1, 2, 3])),
-            repeats=Index([0, 0, 1, 1, 2, 2]),
-        )
-        self.setup_indices()
+    @pytest.fixture
+    def index(self, request):
```

Comment:
I think this is pretty much the only non-standard thing I had to do.  A lot of the tests in this file only tested a subset of the indexes in `indices`, so I needed a way to mimic that with fixtures.  The idea here is to use indirect parametrization to allow this behavior, e.g. 
```python
@pytest.mark.parametrize("index", ["int", "uint", "range"], indirect=True)
```
Would result in this fixture returning the three indexes keyed by `"int"`, `"uint"`, `"range"` in `indices_dict`.

Reply:
> I think this is pretty much the only non-standard thing I had to do

this is OK. used in io.formats.test_to_html for the biggie_df_fixture.

the alternative is to create a fixture yielding just the ids which is then requested by the indices fixture. (see filepath_or_buffer_id and filepath_or_buffer fixtures in io.formats.test_format.) and then override the id fixture directly at function, class or module level.

both patterns (direct and indirect parameterisation) are equivalent.

Reply:
looking at this in detail. I don't think you need to create a separate fixture in order to do this. you could apply indirect parametrisation to the indices fixture.